mhep-transition/mhep-django
# Point Emoncms at Django API endpoints

This merge points the emoncms frontend at the Django endpoints. It has been tested to ensure that a user can 

It has been manually tested to ensure you can:

* Create an assessment (set name & description)
* Change the status of an assessment
* Update assessment (change name & description)
* Delete an assessment
* Save data in assessment (e.g. by adding floors, entering values in UI)
* Create a new scenario
* Add an item from a library e.g. a wall
* From within an assessment:
  * Create new item in library (based on empty & based on copy)
  * Edit an item in a library
  * Delete an item from a library
  * Create a new library (based on empty & based on copy)
* From Libraries manager:
  * Show the items in a library
  * Create new item in library (based on empty & based on copy)
  * Edit an item in a library (savelibrary.json)
  * Delete an item from a library
  * Create a new library (based on empty & based on copy)
  * Edit a library (name)
  * Delete a library
* Add a fuel in Fuels manager
* Change the openBEM version after the assessment is created (default should be 10.1.1)
* Export project data to text box
* Import project data from text box
* Download project data to file
* Upload project data from file

The following capabilities have not been tested, as the first release of the Django backend does not support multiple users, organisations, images, or exporting PDFs:

* Create an organisation
* Create an assessment in an organisation
* Share an assessment
* Do anything with image gallery
* Do anything with PDF reports
* Share library
* Do anything with an account (change profile, log out etc)
* Manage users in Libraries manager

# Re-routing calls

It re-routes the following calls:

## Assessments

* `assessment/list.json`

* `assessment/create`
* `assessment/get`
* `assessment/setnameanddescription`
* `assessment/setopenBEMversion`
* `assessment/setstatus`
* `assessment/setdata`
* `assessment/delete`

## Organisations

* `assessment/getorganisations`
* `assessment/neworganisation` (untested)

We shim the new call to `/organisations` to return the org.id as a key.

## Libraries

* `assessment/loaduserlibraries`
* `assessment/newlibrary`
* `assessment/savelibrary`
* `assessment/copylibrary`
* `assessment/setlibraryname`
* `assessment/deletelibrary`

* `assessment/additemtolibrary`
* `assessment/edititeminlibrary`
* `assessment/deletelibraryitem`

Library persmissions are no longer managed by `libraryHelper.library_permissions`, and instead are defined by an argument (`writeable`) on each item in `libraryHelper.libraries`.

This merge also improves the way the user's library is synchronised with the standard library.

# Notes

These changes meant that in some cases the front-end expected JSON strings, where now we returing JSON dictionaries, so it's no longer necessary to call `JSON.parse` on the results.

The openBEM version is no longer flipped in PHP, and we expect the API to return the correct version.

This merge also disables all other PHP JSON endpoints, making it clearer what else will need to be ported.
